modules/yuktamediaAnalyticsAdapter.js

```diff
@@ -0,0 +1,144 @@
+          bidWon: mapBidResponse(args, 'win')
+        }, 'won');
+      }
+    }
+
+    if (eventType === CONSTANTS.EVENTS.AUCTION_END) {
+      send(events, 'auctionEnd');
+    }
+  }
+});
+
+function mapBidRequests(params) {
+  let arr = [];
+  if (typeof params.bids !== 'undefined' && params.bids.length) {
+    params.bids.forEach(function (bid) {
+      arr.push({
+        bidderCode: bid.bidder,
+        bidId: bid.bidId,
+        adUnitCode: bid.adUnitCode,
+        requestId: bid.bidderRequestId,
+        auctionId: bid.auctionId,
+        transactionId: bid.transactionId,
+        sizes: utils.parseSizesInput(bid.mediaTypes.banner.sizes).toString(),
+        renderStatus: 1,
+        requestTimestamp: params.auctionStart
+      });
+    });
+  }
+  return arr;
+}
+
+function mapBidResponse(bidResponse, status) {
+  if (status !== 'win') {
+    let bid = events.bids.filter(o => o.bidId == bidResponse.bidId || o.bidId == bidResponse.requestId)[0];
+    Object.assign(bid, {
+      bidderCode: bidResponse.bidder,
+      bidId: status == 'timeout' ? bidResponse.bidId : bidResponse.requestId,
+      adUnitCode: bidResponse.adUnitCode,
+      auctionId: bidResponse.auctionId,
+      creativeId: bidResponse.creativeId,
+      transactionId: bidResponse.transactionId,
+      currency: bidResponse.currency,
+      cpm: bidResponse.cpm,
+      netRevenue: bidResponse.netRevenue,
+      mediaType: bidResponse.mediaType,
+      statusMessage: bidResponse.statusMessage,
+      status: bidResponse.status,
+      renderStatus: status == 'timeout' ? 3 : 2,
+      timeToRespond: bidResponse.timeToRespond,
+      requestTimestamp: bidResponse.requestTimestamp,
+      responseTimestamp: bidResponse.responseTimestamp
+    });
+  } else if (status == 'win') {
+    return {
+      bidderCode: bidResponse.bidder,
+      bidId: bidResponse.requestId,
+      adUnitCode: bidResponse.adUnitCode,
+      auctionId: bidResponse.auctionId,
+      creativeId: bidResponse.creativeId,
+      transactionId: bidResponse.transactionId,
+      currency: bidResponse.currency,
+      cpm: bidResponse.cpm,
+      netRevenue: bidResponse.netRevenue,
+      renderedSize: bidResponse.size,
+      mediaType: bidResponse.mediaType,
+      statusMessage: bidResponse.statusMessage,
+      status: bidResponse.status,
+      renderStatus: 4,
+      timeToRespond: bidResponse.timeToRespond,
+      requestTimestamp: bidResponse.requestTimestamp,
+      responseTimestamp: bidResponse.responseTimestamp
+    }
+  }
+}
+
+function send(data, status) {
+  let location = utils.getWindowLocation();
+  let secure = location.protocol == 'https:';
+  if (typeof data !== 'undefined' && typeof data.auctionInit !== 'undefined') {
+    data.auctionInit = Object.assign({ host: location.host, path: location.pathname, search: location.search }, data.auctionInit);
+  }
+  data.initOptions = initOptions;
+
+  let yuktamediaAnalyticsRequestUrl = url.format({
+    protocol: secure ? 'https' : 'http',
```

Comment:
As part of Prebid 3.x - all requests need to be made with `https`.  Can you update this logic here to hard-code the protocol value?

Reply:
Hi @jsnellbaker 

I have update protocol to https as per requirement and your suggestion. Can you please re-review and merge into master.

Thanks.